UCI-HyperXite/hx9-control-system
# Set up basic Socket.IO server and client

Add Socket.IO client to control station and socketioxide server to pod operation.
Closes #3 and #4.

## Changes
- Add basic socketioxide server example
  - Create handler function on `control-station` namespace
  - Acknowledge "ping" with a "pong" and emit a separate pong
- Install socket.io-client for control station
- Import pod socket client, handler, and hook
  - Import source files from [UCI-HyperXite/hx8-control-system](https://github.com/UCI-HyperXite/hx8-control-system) for managing the Socket.IO connection and state on the control-station
  - Remove old `PodData` attributes and include disconnect handler
- Resolve type issues in socket server events
  - Use `Entries` type and bind `this` to handler references
- Connect socket client with Dashboard and ping/pong
  - Incorporate socket connected state with Dashboard
  - Include reason in disconnect handler
  - Handle acknowledgement during emit ping
  - Configure Vite proxy for socket.io requests

## Testing
1. Run the control station and connect to the local development server (localhost:5173)
2. Observe the Dashboard says **disconnected**
3. Start the socket server by running the rust program in pod-operation
4. Observe the Dashboard says **connected**
5. Open the browser's JavaScript console
6. Press the **Send Ping** button and observe the server acknowledges the ping and sends its own pong
